### PR TITLE
fix: fix unknowns not present in initialization system in linearization

### PR DESCRIPTION
**Pull request by AayushSabharwal**
## Checklist

- [ ] Appropriate tests were added
- [ ] Any code changes were done in a way that does not break public API
- [ ] All documentation related to code changes were updated
- [ ] The new code follows the
  [contributor guidelines](https://github.com/SciML/.github/blob/master/CONTRIBUTING.md), in particular the [SciML Style Guide](https://github.com/SciML/SciMLStyle) and
  [COLPRAC](https://github.com/SciML/COLPRAC).
- [ ] Any new documentation only uses public API

### CODE REVIEW

**Comment by ChrisRackauckas** [comment]
Test the case I described, `x => 2, x ~ y`, and `y => 2, x ~ y`.

**Comment by AayushSabharwal** [comment]
Does this test work?

**Comment by ChrisRackauckas** [comment]
No it needs to be structurally simplified.

File: src/systems/abstractsystem.jl [hunk 1]
```diff
@@ -2002,7 +2002,9 @@ function linearization_function(sys::AbstractSystem, inputs,
                 p = todict(p)
                 newps = deepcopy(sys_ps)
                 for (k, v) in p
-                    setp(sys, k)(newps, v)
+                    if is_parameter(sys, p)
```
Comment (baggepinnen) [inline src/systems/abstractsystem.jl:2005]: what happens if there is something for which `is_parameter(sys, p)`?

Comment (AayushSabharwal) [reply src/systems/abstractsystem.jl:2005]: That's a bug, thanks 😅

Comment (baggepinnen) [reply src/systems/abstractsystem.jl]: Haha, you fixed something other than I had in mind actually, let me rephrase the question after the fix

what happens if there is something for which !is_parameter(sys, k)?

not `p` -> `k`

Comment (AayushSabharwal) [reply src/systems/abstractsystem.jl]: Ah. Well that piece of code handles parameters specified as a `Dict`. Its sole purpose is to update the `MTKParameters` object. If `k` is not a parameter, it's not in the parameter object, so it's ignored. If it's an unknown, it's already taken care of in `u`.
